### PR TITLE
9755 add qa to support detail page

### PR DESCRIPTION
**Pull request by jtmst**
## Description
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/9901

Adds Q&A component to the support_resources_detail_page template

## Testing done
Local testing with test data in tugboat environment.

Steps to test:
- Rebuild local to tugboat instance: https://main-vpsyfavqes4gp7mmxayff3gvxqsntwzz.demo.cms.va.gov/
- navigate to resource and support page: resources/getting-a-gi-bill-extension/
- Observe that data payload reflects new graphQL query and that Q&A content block data is included
- Observe that Q&A sections are rendered on page

## Screenshots
Screen Shot 2022-08-24 at 9 27 43 AM
Screen Shot 2022-08-24 at 9 29 30 AM
Screen Shot 2022-08-24 at 9 28 29 AM

## Acceptance criteria
- [ ] Q&As render on on support resource detail page type pages when data is provided

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by maxx1128 — APPROVED** [review]

**Review by allisonlu — APPROVED** [review]

**Review by ryguyk — CHANGES_REQUESTED** [review]

File: src/site/stages/build/drupal/graphql/paragraph-fragments/qaGroup.paragraph.graphql.js [hunk 1]
```diff
@@ -13,6 +13,16 @@ fragment qaGroup on ParagraphQAGroup {
         moderationState
         ... on NodeQA {
           fieldAnswer {
+              entity {
+                entityBundle
+              ... on ParagraphRichTextCharLimit1000 {
+                fieldWysiwyg {
+                  value
```
[the remark this answers is not in the document]
Comment (jtmst) [reply src/site/stages/build/drupal/graphql/paragraph-fragments/qaGroup.paragraph.graphql.js]: fixed

File: src/site/paragraphs/q_a_group.drupal.liquid [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div data-template="paragraphs/q_a_group" data-entity-id="{{ entity.entityId }}" data-analytics-faq-text="{{ entity.entityLabel | escape }}">
+  <div id="{{ entity.entityId }}">
+    <div data-entity-id="{{ entity.entityId }}">
+        {% assign bundleComponent = "src/site/paragraphs/" | append: "rich_text_char_limit_1000" | append: ".drupal.liquid" %}
```
[the remark this answers is not in the document]
Comment (jtmst) [reply src/site/paragraphs/q_a_group.drupal.liquid]: good call, fixed

**Review by ryguyk — APPROVED** [review]

File: src/site/layouts/support_resources_detail_page.drupal.liquid [hunk 3]
```diff
@@ -47,6 +47,35 @@
...
             {% endfor %}
 
+            <!-- QAs -->
+          {% for contentBlock in fieldContentBlock %}
+            {% if contentBlock.entity.entityBundle === "q_a_group"  %}
+              {% assign fieldQAs = contentBlock.entity.queryFieldQAs.entities %}
+
+              <!-- Optional section header -->
+              {% if contentBlock.entity.fieldSectionHeader %}
+                <button type="button" class="vads-u-visibility--screen-reader">Back to top</button>
```
Comment (timwright12) [inline src/site/layouts/support_resources_detail_page.drupal.liquid:57]:
```suggestion
                <button type="button" class="vads-u-visibility--screen-reader" type="button">Back to top</button>
```

File: src/site/layouts/support_resources_detail_page.drupal.liquid [hunk 4]
```diff
@@ -47,6 +47,35 @@
...
+                <h2>{{ contentBlock.entity.fieldSectionHeader }}</h2>
+              {% endif %}
+              
+              {% assign fieldSectionHeaderTag = 'h2' %}
+              {% if contentBlock.entity.fieldSectionHeader %}
+                {% assign fieldSectionHeaderTag = 'h3' %}
+              {% endif %}
+              <button type="button" class="vads-u-visibility--screen-reader">Back to top</button>
```
Comment (timwright12) [inline src/site/layouts/support_resources_detail_page.drupal.liquid:65]:
```suggestion
              <button type="button" class="vads-u-visibility--screen-reader" type="button">Back to top</button>
```

Comment (timwright12) [reply src/site/layouts/support_resources_detail_page.drupal.liquid:65]: Can we use the back to top component here https://design.va.gov/components/back-to-top

File: src/site/layouts/support_resources_detail_page.drupal.liquid (hunk 3 above)
Comment (ndouglas) [reply src/site/layouts/support_resources_detail_page.drupal.liquid:57]: Why the second `type="button"`?

File: src/site/layouts/support_resources_detail_page.drupal.liquid (hunk 4 above)
Comment (jtmst) [reply src/site/layouts/support_resources_detail_page.drupal.liquid]: fixed

**Review by timwright12 — APPROVED** [review]

**Pull request merged**